checks/raw/branch_protection.go

```diff
@@ -72,7 +73,7 @@ func BranchProtection(c clients.RepoClient) (checker.BranchProtectionsData, erro
 		}
 
 		// TODO: if this is a sha, get the associated branch. for now, ignore.
-		if commit.MatchString(release.TargetCommitish) {
+		if commit.MatchString(release.TargetCommitish) && !strings.Contains(c.URI(), "gitlab.") {
```

Comment:
This code calls `c.URI()` indirectly through [here](https://github.com/N8BWert/scorecard/blob/gitlab_tests/checks/branch_protection_test.go#L420) and [here](https://github.com/N8BWert/scorecard/blob/dc3476404834d9b7aba842fbbd628f9635ad8cbc/checks/branch_protection.go#L37).

The mockclient will need to be told to expect a `URI` call
https://github.com/N8BWert/scorecard/blob/gitlab_tests/checks/branch_protection_test.go#L395

Reply:
Since the unit tests haven't previously care about this value, I'm inclined to think an empty value would work, but would need testing.